File: test-fixtures/codebase/Sample.java
package sample;

/**
 * Sample Java class for testing.
 * Demonstrates various Java features including annotations.
 */
public class Sample {
    private String name;

    /**
     * Creates a new Sample instance.
     * @param name the name to use
     */
    public Sample(String name) {
        this.name = name;
    }

    /**
     * Returns a greeting message.
     * @return the greeting string
     */
    @Override
    public String toString() {
        return "Hello, " + name + "!";
    }

    /**
     * Gets the name.
     */
    public String getName() {
        return name;
    }

    private void internalMethod() {
        // Private method for internal use
    }
}

/**
 * Utility class with static methods.
 */
@Deprecated
class PrivateHelper {
    static int calculate(int a, int b) {
        return a + b;
    }

    private static void internalHelper() {
        // Private static method
    }
}

/**
 * Public interface for greeting.
 */
public interface Greeter {
    /**
     * Returns a greeting.
     */
    String greet();
}

/**
 * Add two numbers together.
 */
class MathUtils {
    /**
     * Adds two integers.
     * @param a first number
     * @param b second number
     * @return the sum
     */
    @SuppressWarnings("unused")
    public static int add(int a, int b) {
        return a + b;
    }
}

File: test-fixtures/codebase/sample.rs
//! Sample Rust module for testing.
//!
//! This module demonstrates various Rust features.

/// A greeter struct that holds a name.
/// Used for generating greeting messages.
#[derive(Debug, Clone)]
pub struct Greeter {
    name: String,
}

impl Greeter {
    /// Creates a new Greeter instance.
    pub fn new(name: &str) -> Self {
        Greeter {
            name: name.to_string(),
        }
    }

    /// Returns a greeting message.
    pub fn greet(&self) -> String {
        format!("Hello, {}!", self.name)
    }

    fn private_method(&self) -> &str {
        &self.name
    }
}

/// A private struct for internal use.
struct PrivateHelper {
    value: i32,
}

impl PrivateHelper {
    fn calculate(&self, x: i32) -> i32 {
        self.value + x
    }
}

/// Add two numbers together.
/// Returns the sum of a and b.
#[inline]
pub fn add(a: i32, b: i32) -> i32 {
    a + b
}

fn private_function() -> i32 {
    42
}

/// Process data with validation.
#[must_use]
pub fn process_data(data: &str) -> Option<String> {
    if data.is_empty() {
        None
    } else {
        Some(data.to_uppercase())
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_add() {
        assert_eq!(add(2, 2), 4);
    }

    #[test]
    fn test_greeter() {
        let g = Greeter::new("World");
        assert_eq!(g.greet(), "Hello, World!");
    }
}
